### PR TITLE
Add a drmemtrace analysis tool for syscall mix

### PR DESCRIPTION
**Pull request by abhinav92003**
Adds a new drmemtrace analysis tool (syscall_mix) that counts the frequency of system calls in the trace. It uses the recently added TRACE_MARKER_TYPE_SYSCALL markers to count system calls.

Adds unit tests that invoke syscall_mix in online and offline modes.

### CODE REVIEW

File: clients/drcachesim/tools/syscall_mix.cpp [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+bool
+syscall_mix_t::parallel_shard_memref(void *shard_data, const memref_t &memref)
+{
+    shard_data_t *shard = reinterpret_cast<shard_data_t *>(shard_data);
+    if (memref.marker.type != TRACE_TYPE_MARKER ||
+        memref.marker.marker_type != TRACE_MARKER_TYPE_SYSCALL) {
+        return true;
+    }
+    ++shard->syscall_counts[static_cast<int>(memref.marker.marker_value)];
```
Comment (derekbruening) [inline clients/drcachesim/tools/syscall_mix.cpp:102]: assert there's nothing in the upper bits for x64

File: clients/drcachesim/tools/syscall_mix_create.h [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ */
+
+/**
+ * Creates an analysis tool which counts the number of instances of each system call
+ * in the trace.
+ */
+analysis_tool_t *
+syscall_mix_tool_create(unsigned int verbose = 0);
+
+#endif /* _OPCODE_MIX_CREATE_H_ */
```
Comment (derekbruening) [inline clients/drcachesim/tools/syscall_mix_create.h:52]: Stale comment

File: suite/tests/CMakeLists.txt [hunk 3]
```diff
@@ -4214,6 +4214,10 @@ if (BUILD_CLIENTS)
         "@-simulator_type@basic_counts" "")
     endif (NOT WIN32)
 
+    # Test syscall_mix.
+    torunonly_drcachesim(syscall-mix ${ci_shared_app} "-simulator_type syscall_mix" "")
+    torunonly_drcacheoff(syscall-mix ${ci_shared_app} "" "@-simulator_type@syscall_mix" "")
+
```
Comment (derekbruening) [inline suite/tests/CMakeLists.txt:4220]: Add a section to sec_drcachesim_tools with sample invocation and output

Comment (derekbruening) [inline suite/tests/CMakeLists.txt:4220]: Add to the library list in sec_drcachesim_newtool

Comment (derekbruening) [inline suite/tests/CMakeLists.txt:4220]: Add to the changelist in release.dox

**Review by derekbruening — APPROVED** [review]

File: clients/drcachesim/tools/syscall_mix_create.h (hunk 2 above)
Comment (abhinav92003) [reply clients/drcachesim/tools/syscall_mix_create.h:52]: Done.

File: suite/tests/CMakeLists.txt (hunk 3 above)
Comment (abhinav92003) [reply suite/tests/CMakeLists.txt:4220]: Done.

Comment (abhinav92003) [reply suite/tests/CMakeLists.txt:4220]: Done.

Comment (abhinav92003) [reply suite/tests/CMakeLists.txt:4220]: Done.

**Pull request merged**